cs/src/Management/TunnelManagementClient.cs

```diff
@@ -915,41 +829,6 @@ public void Dispose()
             return authHeader;
         }
 
-        private async Task<AuthenticationHeaderValue?> GetAuthenticationHeaderAsync(
-            TunnelV2? tunnel,
-            string[]? accessTokenScopes,
-            TunnelRequestOptions? options)
-        {
-            AuthenticationHeaderValue? authHeader = null;
-
-            if (!string.IsNullOrEmpty(options?.AccessToken))
-            {
-                TunnelAccessTokenProperties.ValidateTokenExpiration(options.AccessToken);
-                authHeader = new AuthenticationHeaderValue(
-                    TunnelAuthenticationScheme, options.AccessToken);
-            }
-
-            if (authHeader == null)
-            {
-                authHeader = await this.userTokenCallback();
-            }
-
-            if (authHeader == null && tunnel?.AccessTokens != null && accessTokenScopes != null)
-            {
-                foreach (var scope in accessTokenScopes)
-                {
-                    if (tunnel.TryGetValidAccessToken(scope, out string? accessToken))
-                    {
-                        authHeader = new AuthenticationHeaderValue(
-                            TunnelAuthenticationScheme, accessToken);
-                        break;
-                    }
-                }
-            }
-
-            return authHeader;
-        }
-
         /// <inheritdoc />
         public async Task<Tunnel[]> ListTunnelsAsync(
```

Comment:
New Azure API's result for list tunnels will be `TunnelListByRegionResponse` with my recent PR. It'll need to be converted to `Tunnel[]` be enumerating over the regions in `Value` with `SelectMany`.

Reply:
updated